fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/metadata/RecordMetaDataBuilderTest.java

```diff
@@ -93,52 +108,62 @@ public void caching(final TestHelpers.BooleanEnum deprecatedWay) throws Exceptio
         RecordMetaData metaData = builder.getRecordMetaData();
         Index index = metaData.getIndex("MySimpleRecord$PRIMARY");
         assertNotNull(index);
         assertNotNull(index.getPrimaryKeyComponentPositions());
         assertArrayEquals(new int[] {0}, index.getPrimaryKeyComponentPositions());
     }
 
-    @EnumSource(TestHelpers.BooleanEnum.class)
-    @ParameterizedTest(name = "caching [deprecatedWay = {0}]")
-    public void indexOnNestedPrimaryKey(final TestHelpers.BooleanEnum deprecatedWay) throws Exception {
+    @MethodSource("indexTestArguments")
+    @ParameterizedTest(name = "caching [deprecatedWay = {0}, indexCounterBasedSubspaceKey = {1}]")
+    public void indexOnNestedPrimaryKey(final TestHelpers.BooleanEnum deprecatedWay, final TestHelpers.BooleanEnum indexCounterBasedSubspaceKey) throws Exception {
         RecordMetaDataBuilder builder = createBuilder(TestRecordsWithHeaderProto.getDescriptor(), deprecatedWay.toBoolean());
         builder.getRecordType("MyRecord")
                 .setPrimaryKey(field("header").nest("rec_no"));
         builder.addIndex("MyRecord", new Index("MyRecord$PRIMARY",
                 field("header").nest("rec_no"),
                 IndexTypes.VALUE));
+        if (!deprecatedWay.toBoolean() && indexCounterBasedSubspaceKey.toBoolean()) {
+            builder.enableIndexCounterBasedSubspaceKey();
+        }
         RecordMetaData metaData = builder.getRecordMetaData();
         Index index = metaData.getIndex("MyRecord$PRIMARY");
         assertNotNull(index);
         assertNotNull(index.getPrimaryKeyComponentPositions());
         assertArrayEquals(new int[] {0}, index.getPrimaryKeyComponentPositions());
     }
 
-    @EnumSource(TestHelpers.BooleanEnum.class)
-    @ParameterizedTest(name = "caching [deprecatedWay = {0}]")
-    public void indexOnPartialNestedPrimaryKey(final TestHelpers.BooleanEnum deprecatedWay) throws Exception {
+    @MethodSource("indexTestArguments")
+    @ParameterizedTest(name = "caching [deprecatedWay = {0}, indexCounterBasedSubspaceKey = {1}]")
+    public void indexOnPartialNestedPrimaryKey(final TestHelpers.BooleanEnum deprecatedWay, final TestHelpers.BooleanEnum indexCounterBasedSubspaceKey) throws Exception {
         RecordMetaDataBuilder builder = createBuilder(TestRecordsWithHeaderProto.getDescriptor(), deprecatedWay.toBoolean());
         builder.getRecordType("MyRecord")
                 .setPrimaryKey(field("header").nest(concatenateFields("path", "rec_no")));
         builder.addIndex("MyRecord", new Index("MyRecord$path_str",
                 concat(field("header").nest("path"),
                         field("str_value")),
                 IndexTypes.VALUE));
+        if (!deprecatedWay.toBoolean() && indexCounterBasedSubspaceKey.toBoolean()) {
+            builder.enableIndexCounterBasedSubspaceKey();
+        }
         RecordMetaData metaData = builder.getRecordMetaData();
         Index index = metaData.getIndex("MyRecord$path_str");
```

Comment:
Presumably, this should check the subspace key if the counter-based strategy is used? I don't think there are any tests that make sure that the subspace key is actually set.

Reply:
sure.